Userland/Libraries/LibWeb/Painting/PaintableBox.cpp

```diff
@@ -406,6 +406,12 @@ BorderRadiiData PaintableBox::normalized_border_radii_data(ShrinkRadiiForBorders
 
 Optional<CSSPixelRect> PaintableBox::calculate_overflow_clipped_rect() const
 {
+    if (layout_node().dom_node() && layout_node().dom_node()->is_document()) {
+        // NOTE: We ignore document's box because it has size of viewport and missing scroll offset which
+        //       means we can't find its resulting position and combine with clip rects of other boxes
+        return {};
+    }
```

Comment:
It feels a little confusing to say that document's box has the size of the viewport.
The document's box *is* the `Layout::Viewport` :sweat_smile: 
I think we can instead just check `if (layout_node().is_viewport())` here.